src/components/Icon/tests/Icon.test.tsx

```diff
@@ -1,12 +1,280 @@
+        it('basic icon should have rendered one span element', () => {
+          const iconWrapper = mount(
+                                    <Icon source="placeholder" theme={theme} />,
+                              );
+          expect(iconWrapper.find('span')).toHaveLength(1);
+        });
+        it('basic icon should have rendered with one icon clss on span', () => {
+          const iconWrapper = mount(
+                                    <Icon source="placeholder" theme={theme} />,
+                              );
+          expect(iconWrapper.find('span').hasClass('Icon'));
+        });
+        it('basic icon should have rendered one div element', () => {
+          const iconWrapper = mount(
+                                    <Icon source="placeholder" theme={theme} />,
+                              );
+          expect(iconWrapper.find('div')).toHaveLength(1);
+        });
+        it('basic icon should have rendered with one placeholder clss on div', () => {
+          const iconWrapper = mount(
+                                    <Icon source="placeholder" theme={theme} />,
+                              );
+          expect(iconWrapper.find('div').hasClass('Placeholder'));
+        });
+        it('should verify given source', () => {
+          const iconWrapper = mount(
+                                    <Icon source="placeholder" theme={theme} />,
+                              );
+          expect(iconWrapper.prop('source')).toBe('placeholder');
+        });
+      });
+  });
+
+  describe('color property', () => {
+    describe('when set', () => {
+      it('basic icon should have rendered one span element', () => {
+        const iconWrapper = mount(
+                                  <Icon source="notes" color="white" theme={theme} />,
+                            );
+        expect(iconWrapper.find('span')).toHaveLength(1);
+      });
+      it('basic icon should have rendered with one colorWhite clss on span', () => {
+        const iconWrapper = mount(
+                                  <Icon source="placeholder" color="white" theme={theme} />,
+                            );
+        expect(iconWrapper.find('span').hasClass('colorWhite'));
+      });
+      it('should verify given color when source as svg ', () => {
+        const iconWrapper = mount(
+                                  <Icon source="notes" color="white" theme={theme} />,
+                            );
+        expect(iconWrapper.prop('color')).toBe('white');
+      });
+      it('should verify given color when source as placeholder ', () => {
+        const iconWrapper = mount(
+                                  <Icon source="placeholder" color="white" theme={theme} />,
+                            );
+        expect(iconWrapper.prop('color')).toBe('white');
+      });
+    });
+
+    describe('when not set', () => {
+      it('basic icon should have rendered one span element', () => {
+        const iconWrapper = mount(
```

Comment:
Thanks for highlighting this.I have updated unit tests as per your comments and removed redundant icon wrapper from all it statements.There was no reason to do this way.